enigma-principal/enclave/src/epoch_keeper_t/mod.rs

```diff
@@ -243,7 +243,27 @@ pub mod tests {
             stakes: stakes.into_iter().map(|s| U256::from(s.clone())).collect(),
         };
         let epoch2 = Epoch { nonce: U256::from(1), seed: U256::from(46661), worker_params: worker_params2 };
-        let image2 = epoch2.raw_encode();
-        assert_eq!(image2, reference_image_hex2.from_hex::<Vec<u8>>().unwrap());
+        let image2 = epoch2.encode_epoch();
+        assert_eq!(image2, expected_image2);
+    }
+
+    fn create_epoch() -> Epoch {
+        let block_number = U256::from(67);
+        let workers = vec![H160::from_slice(&[1u8;20]), H160::from_slice(&[2u8;20]), H160::from_slice(&[3u8;20])];
+        let stakes = vec![U256::from(20), U256::from(30), U256::from(40)];
+
+        let worker_params = InputWorkerParams {block_number, workers, stakes};
+        let nonce = U256::from(555);
+        let seed = U256::from(666);
+        Epoch { nonce, seed, worker_params }
+    }
+
+    fn get_expected_encoded_epoch() -> Vec<u8> {
+        [1, 0, 0, 0, 0, 0, 0, 0, 220, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 154, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 43, 1, 0, 0, 0, 0, 0, 0, 0, 60, 0, 0, 0, 0, 0, 0, 0, 0, 20, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 20, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0, 20, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 1, 0, 0, 0, 0, 0, 0, 0, 96, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 20, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 30, 0, 0, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 40].to_vec()
+    }
+    pub fn test_raw_encode() {
```

Comment:
What does this test add beyond `test_create_epoch_image`?

Reply:
removed it.